### PR TITLE
[RFR] Fix FileInput documentation

### LINKED ISSUE #2151: FileInput doesn't accept multiple files when setting 'multiple' property using options prop
**What you were expecting:**
A FileInput that accepts multiple files.

**What happened instead:**
The input only accepts a single file.

**Steps to reproduce:**
Set multiple using the `options` prop rather than the `multiple` prop.

From [the docs](https://github.com/marmelab/react-admin/blob/master/docs/Inputs.md#fileinput):
> The FileInput component accepts an options prop into which you can pass all the react-dropzone properties. For instance, if you need to upload several files at once, just add the options={{ multiple: true }} DropZone attribute to your <FileInput /> field.

**Related code:**
```
<FileInput source="files" options={{ multiple: true }}>
    <FileField source="src" title="title" />
</FileInput>
```

**Other information:**
I will create a pull request for FileInput to check the options prop in addition to the (legacy?) multiple prop. Otherwise, we should update the documentation to mention the multiple prop.

**Environment**
* React-admin version: 2.2.0

### PR DESCRIPTION
**Pull request by lihop**
Closes #2151

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/FileInput.js [hunk 1]
```diff
@@ -75,13 +76,15 @@ export class FileInput extends Component {
     }
 
     onDrop = files => {
-        const updatedFiles = this.props.multiple
+        const multiple = this.props.options.multiple || this.props.multiple;
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/FileInput.js]: I think we should support only one of the two. As it is a very common usecase, I wonder if it should be `this.props.multiple` in which case, we should merge it in the options before passing them to the `DropZone` component. This should also be documented if chosen.

Comment (djhi) [reply packages/ra-ui-materialui/src/input/FileInput.js]: The code must be updated as well. We currently check for `this.props.multiple` but it should be `this.props.options.multiple`

Comment (djhi) [reply packages/ra-ui-materialui/src/input/FileInput.js]: Nvm, we actually pass `this.props.multiple` to the DropZone...

**Comment by lihop** [comment]
Closed in favour of #2235.

**Pull request closed without merging**